### PR TITLE
Updated matplotlib version requirements

### PR DESCRIPTION
**Pull request by tobyLovick**
# Description

If installed with matplotlib 3.4.3, anesthetic is unable to run plot_2d on anything. The final error is 'AttributeError: 'OffDiagonalAxes' object has no attribute '_shared_axes'. This is fixed by requiring matplotlib 3.5 (Which this pull request implements)
 
```python 
AttributeError                            Traceback (most recent call last)
~\AppData\Local\Temp/ipykernel_9708/2928473168.py in <module>
      4 params = ["x0","x1"]
      5 fig, axes = make_2d_axes(params, figsize=(6, 6), facecolor='w')
----> 6 prior.plot_2d(axes, alpha=0.9, label="prior")
      7 samples.plot_2d(axes, alpha=0.9, label="posterior")
      8 axes.iloc[-1, 0].legend(bbox_to_anchor=(len(axes)/2, len(axes)), loc='lower center', ncols=2)

~\anaconda3\lib\site-packages\anesthetic\samples.py in plot_2d(self, axes, *args, **kwargs)
    354                             ax.set_ylabel(ylabel)
    355                         else:
--> 356                             self.plot(x, y, ax=ax, xlabel=xlabel,
    357                                       ylabel=ylabel, *args, **lkwargs)
    358                             ax.set_xlabel(xlabel)

~\anaconda3\lib\site-packages\anesthetic\plotting\_core.py in __call__(self, *args, **kwargs)
     74                 "samples.plot.kde_2d(x=x, y=y, ax=ax)"
     75                 )
---> 76         return super().__call__(*args, **kwargs)

~\anaconda3\lib\site-packages\pandas\plotting\_core.py in __call__(self, *args, **kwargs)
    943         if kind in self._dataframe_kinds:
    944             if isinstance(data, ABCDataFrame):
--> 945                 return plot_backend.plot(data, x=x, y=y, kind=kind, **kwargs)
    946             else:
    947                 raise ValueError(f"plot kind {kind} can only be used for data frames")

~\anaconda3\lib\site-packages\pandas\plotting\_matplotlib\__init__.py in plot(data, kind, **kwargs)
     69             kwargs["ax"] = getattr(ax, "left_ax", ax)
     70     plot_obj = PLOT_CLASSES[kind](data, **kwargs)
---> 71     plot_obj.generate()
     72     plot_obj.draw()
     73     return plot_obj.result

~\anaconda3\lib\site-packages\pandas\plotting\_matplotlib\core.py in generate(self)
    453         self._add_table()
    454         self._make_legend()
--> 455         self._adorn_subplots()
    456 
    457         for ax in self.axes:

~\anaconda3\lib\site-packages\pandas\plotting\_matplotlib\core.py in _adorn_subplots(self)
    677             all_axes = self._get_subplots()
    678             nrows, ncols = self._get_axes_layout()
--> 679             handle_shared_axes(
    680                 axarr=all_axes,
    681                 nplots=len(all_axes),

~\anaconda3\lib\site-packages\pandas\plotting\_matplotlib\tools.py in handle_shared_axes(axarr, nplots, naxes, nrows, ncols, sharex, sharey)
    417                         continue
    418                     if sharex or _has_externally_shared_axis(ax, "x"):
--> 419                         _remove_labels_from_axis(ax.xaxis)
    420 
    421             except IndexError:

~\anaconda3\lib\site-packages\pandas\plotting\_matplotlib\tools.py in _remove_labels_from_axis(axis)
    319 
    320 def _remove_labels_from_axis(axis: Axis):
--> 321     for t in axis.get_majorticklabels():
    322         t.set_visible(False)
    323 

~\anaconda3\lib\site-packages\matplotlib\axis.py in get_majorticklabels(self)
   1182     def get_majorticklabels(self):
   1183         """Return this Axis' major tick labels, as a list of `~.text.Text`."""
-> 1184         ticks = self.get_major_ticks()
   1185         labels1 = [tick.label1 for tick in ticks if tick.label1.get_visible()]
   1186         labels2 = [tick.label2 for tick in ticks if tick.label2.get_visible()]

~\anaconda3\lib\site-packages\matplotlib\axis.py in get_major_ticks(self, numticks)
   1353         r"""Return the list of major `.Tick`\s."""
   1354         if numticks is None:
-> 1355             numticks = len(self.get_majorticklocs())
   1356 
   1357         while len(self.majorTicks) < numticks:

~\anaconda3\lib\site-packages\matplotlib\axis.py in get_majorticklocs(self)
   1258     def get_majorticklocs(self):
   1259         """Return this Axis' major tick locations in data coordinates."""
-> 1260         return self.major.locator()
   1261 
   1262     def get_minorticklocs(self):

~\anaconda3\lib\site-packages\matplotlib\ticker.py in __call__(self)
   2261 
   2262     def __call__(self):
-> 2263         vmin, vmax = self.axis.get_view_interval()
   2264         return self.tick_values(vmin, vmax)
   2265 

~\anaconda3\lib\site-packages\matplotlib\axis.py in getter(self)
   1987     def getter(self):
   1988         # docstring inherited.
-> 1989         return getattr(getattr(self.axes, lim_name), attr_name)
   1990 
   1991     def setter(self, vmin, vmax, ignore=False):

~\anaconda3\lib\site-packages\matplotlib\axes\_base.py in viewLim(self)
    780     @property
    781     def viewLim(self):
--> 782         self._unstale_viewLim()
    783         return self._viewLim
    784 

~\anaconda3\lib\site-packages\matplotlib\axes\_base.py in _unstale_viewLim(self)
    776             for ax in self._shared_y_axes.get_siblings(self):
    777                 ax._stale_viewlim_y = False
--> 778             self.autoscale_view(scalex=scalex, scaley=scaley)
    779 
    780     @property

~\anaconda3\lib\site-packages\matplotlib\axes\_base.py in autoscale_view(self, tight, scalex, scaley)
   2751             # End of definition of internal function 'handle_single_axis'.
   2752 
-> 2753         handle_single_axis(
   2754             scalex, self._autoscaleXon, self._shared_x_axes, 'intervalx',
   2755             'minposx', self.xaxis, self._xmargin, x_stickies, self.set_xbound)

~\anaconda3\lib\site-packages\matplotlib\axes\_base.py in handle_single_axis(scale, autoscaleon, shared_axes, interval, minpos, axis, margin, stickies, set_bound)
   2748             if not self._tight:
   2749                 x0, x1 = locator.view_limits(x0, x1)
-> 2750             set_bound(x0, x1)
   2751             # End of definition of internal function 'handle_single_axis'.
   2752 

~\anaconda3\lib\site-packages\matplotlib\axes\_base.py in set_xbound(self, lower, upper)
   3396             upper = old_upper
   3397 
-> 3398         self.set_xlim(sorted((lower, upper),
   3399                              reverse=bool(self.xaxis_inverted())),
   3400                       auto=None)

~\anaconda3\lib\site-packages\anesthetic\plot.py in set_xlim(self, left, right, emit, auto, xmin, xmax)
    264                     self.callbacks.process('xlim_changed', self)
    265                     # Call all other x-axes that are shared with this one
--> 266                     for other in self._shared_axes['x'].get_siblings(self):
    267                         if other is not self:
    268                             other.set_xlim(left, right, emit=False, auto=auto)

AttributeError: 'OffDiagonalAxes' object has no attribute '_shared_axes' 
``` 

Fixes # (issue)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [ ] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [ ] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have appropriately incremented the [semantic version number](https://semver.org/) in both README.rst and anesthetic/_version.py

### CODE REVIEW

**Comment by williamjameshandley** [comment]
Many thanks @tobyLovick -- I've pushed a couple of additions 
1. A script which gets the dependencies with the version numbers `bin/dependencies.py`
2. An extra CI test which then installs these dependencies

**Comment by williamjameshandley** [comment]
I have now changed `>=` to `==` and reverted to an earlier version of matplotlib which should fail. Once that does, could you update the version numbers to be in sync with pandas (e.g. we don't need to require a min version of scipy since pandas requires a higher one anyway).

**Comment by williamjameshandley** [comment]
Hi @tobyLovick, it looks like the builds are failing because it turns out [pandas](https://pandas.pydata.org/docs/getting_started/install.html#dependencies) only has scipy as an [optional dependency](https://pandas.pydata.org/docs/getting_started/install.html#dependencies). We should therefore specify scipy as a dependency in the `pyproject.toml`, without a minimum version (until we find we need to specify one).

File: pyproject.toml [hunk 1]
```diff
@@ -23,10 +23,10 @@ readme = "README.rst"
 license = {file = "LICENSE"}
 requires-python = ">=3.8"
 dependencies = [
-    "scipy>=1.2.0",
-    "numpy",
+    "scipy",
+ "numpy",
```
Comment (williamjameshandley) [inline pyproject.toml:27]: In the interests of historical completeness, could you put the (space-based) indentation back in?

**Review by williamjameshandley — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (appetrosyan) [reply pyproject.toml:27]: As someone who is a hard-tab enthusiast, I’m afraid this is a requirement and not just for historical completeness. The Python parsers are subject to the same limitations as the language, so mixing tabs and spaces is a hard error.

Comment (tobyLovick) [reply pyproject.toml:27]: Indentation is added back (with 3 spaces that seemed correct and doing a tab was too big of an indentation?)

Comment (williamjameshandley) [reply pyproject.toml:27]: did you push this update?

**Pull request merged**